sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -0,0 +1,165 @@
+    """Kernel ridge regression and Kernel ridge classification
+    (also known as Kernel Extreme Learning Machine).
+    """
+    # By default, it works as a regressor
+    regression = True
+
+    def __init__(self, alpha=1, kernel="rbf", gamma=None, degree=3, coef0=1,
+                 kernel_params=None):
+        self.alpha = alpha
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+
+    def _get_kernel(self, X, Y=None):
+        if callable(self.kernel):
+            params = self.kernel_params or {}
+        else:
+            params = {"gamma": self.gamma,
+                      "degree": self.degree,
+                      "coef0": self.coef0}
+        return pairwise_kernels(X, Y, metric=self.kernel,
+                                filter_params=True, **params)
+
+    @property
+    def _pairwise(self):
+        return self.kernel == "precomputed"
+
+    def fit(self, X, y=None, sample_weight=None):
+        """Fit Kernel Ridge regression model
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training data
+
+        y : array-like, shape = [n_samples] or [n_samples, n_targets]
+            Target values
+
+        sample_weight : float or array-like of shape [n_samples]
+            Individual weights for each sample, ignored if None is passed.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        # Convert data
+        if sample_weight is not None and not isinstance(sample_weight, float):
+            sample_weight = check_array(sample_weight, ensure_2d=False)
+
+        K = self._get_kernel(X)
+        alpha = np.atleast_1d(self.alpha)
+
+        copy = self.kernel == "precomputed"
+
+        if self.regression is True:
+            X, y = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                             multi_output=True, y_numeric=True)
+            ravel = False
+            if len(y.shape) == 1:
+                y = y.reshape(-1, 1)
+                ravel = True
+
+            self.dual_coef_ = _solve_cholesky_kernel(K, y, alpha,
+                                                     sample_weight,
+                                                     copy)
+            if ravel:
+                self.dual_coef_ = self.dual_coef_.ravel()
+
+        else:
+            # Check alpha
+            if self.alpha <= 0:
+                raise ValueError('C must be positive')
+            X, y = check_X_y(X, y)
+            self.classes_ = unique_labels(y)
+
+            # y must be encoded by zero arrays with 1 in a position
+            # assigned to a label
+            n = X.shape[0]
+            if self.classes_.shape[0] == 1:
+                self.n_classes_ = int(self.classes_[0] + 1)
+                self.classes_ = np.arange(self.n_classes_)
```

Comment:
self.classes_ needs to encode all that was in `y` in the input. It's your job as the estimator developer to convert back to that, not to alter what the user provides.

Reply:
You're right, this will be fixed in the next commit